### PR TITLE
Add support for integrating slides into minutes

### PR DESCRIPTION
**Pull request by dontcallmedom**
This adds the ability to use `Slideset: <url>` followed by `[Slide X]` commands in IRC to display the said slides as part of the generated minutes.

This relies on the `<i-slide>` Web component, an in-progress work to display PDF and a subset of HTML slides.

### CODE REVIEW

**Comment by dontcallmedom** [comment]
cc @tidoust who worked with me on this

File: scribe.perl [hunk 1]
```diff
@@ -141,11 +143,15 @@
 my $stylesheet;			# URL of style sheet, undef = use defaults
 my $mathjax =			# undef = no math; string is MathJax URL
   'https://www.w3.org/scripts/MathJax/3/es5/mml-chtml.js';
+my $islide =			# undef = no math; string is MathJax URL
+  'https://tidoust.github.io/i-slide/i-slide.js';
```
Comment (dontcallmedom) [inline scribe.perl:147]:
```suggestion
  'https://w3c.github.io/i-slide/i-slide.js';
```

File: scribe.perl [hunk 2]
```diff
@@ -1201,7 +1225,10 @@ ($$)
   n => "<p class=anchor id=\"%2\$s\"><a href=\"#%2\$s\">⚓</a></p>\n",
   u => "<p id=%2\$s class=issue><strong>Issue:</strong> %3\$s</p>\n",
   T => "<h4 id=%2\$s>%3\$s</h4>\n",
-  t => "</section>\n\n<section>\n<h3 id=%2\$s>%3\$s</h3>\n");
+  t => "</section>\n\n<section>\n<h3 id=%2\$s>%3\$s</h3>\n",
+  slideset => "<p class=summary>Slideset: %3\$s</p>",
+  slide => "<i-slide src=\"%2\$s\"><p class=summary>[ <a href=\"%2\$s\">Slide %3\$s</a> ]</p></i-slide>"
```
Comment (dontcallmedom) [inline scribe.perl:1230]:
```suggestion
  slide => "<i-slide src=\"%2\$s\" width=600><p class=summary>[ <a href=\"%2\$s\">Slide %3\$s</a> ]</p></i-slide>"
```

File: scribe.perl [hunk 3]
```diff
@@ -141,11 +143,15 @@
 my $stylesheet;			# URL of style sheet, undef = use defaults
 my $mathjax =			# undef = no math; string is MathJax URL
   'https://www.w3.org/scripts/MathJax/3/es5/mml-chtml.js';
+my $islide =			# undef = no math; string is MathJax URL
+  'https://w3c.github.io/i-slide/i-slide.js';
```
Comment (dontcallmedom) [inline scribe.perl:147]:
```suggestion
  'https://w3c.github.io/i-slide/i-slide-1.js';
```

File: tests/slide-1.test [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+<foo> Francois: this rocks
+<foo> [Slide 2]
+<foo> Dom: I'm unconvinced
+EOF
+
+perl scribe.perl -embed $TMP1 >$TMP2 || exit 1
+cat $TMP2
+
+grep -E '<i-slide src="https://example.org/#1">' $TMP2 || exit 1
+grep -E '<script type="module" src="https://tidoust.github.io/i-slide/i-slide.js' $TMP2 || exit 1
```
Comment (dontcallmedom) [inline tests/slide-1.test:21]:
```suggestion
grep -E '<script type="module" src="https://w3c.github.io/i-slide/i-slide-1.js' $TMP2 || exit 1
```

File: scribe.perl [hunk 5]
```diff
@@ -141,11 +143,15 @@
 my $stylesheet;			# URL of style sheet, undef = use defaults
 my $mathjax =			# undef = no math; string is MathJax URL
   'https://www.w3.org/scripts/MathJax/3/es5/mml-chtml.js';
+my $islide =			# undef = no math; string is MathJax URL
```
Comment (dontcallmedom) [inline scribe.perl:146]:
```suggestion
my $islide =			#  string is i-slide library URL
```

File: scribe.perl (hunk 2 above)
Comment (dontcallmedom) [inline scribe.perl:1230]:
```suggestion
  slide => "<p class=summary><i-slide src=\"%2\$s\">[ <a href=\"%2\$s\">Slide %3\$s</a> ]</i-slide></p>"
```

Comment (dontcallmedom) [reply scribe.perl:1230]: Adapt to i-slide defaulting to be `inline-block` per https://github.com/w3c/i-slide/pull/23#issuecomment-924788404 cc @tidoust

File: scribe.perl [hunk 6]
```diff
@@ -1261,7 +1288,7 @@ ($$)
     "type=\"text/css\" title=\"$_->[1]\" href=\"$_->[2]\">"} @stylesheets);
 my $scripts = !$emphasis || !$has_math ? ''
   : "<script src=\"$mathjax\" id=MathJax-script async></script>\n";
-
+$scripts .= ! $has_slideset ? '' : "<script type=\"module\" src=\"$islide\" defer></script>\n";
```
Comment (tidoust) [inline scribe.perl:1291]:
```suggestion
$scripts .= ! $has_slideset ? '' : "<script type=\"module\" src=\"$islide\"></script>\n";
```

See https://github.com/w3c/i-slide/pull/34

**Comment by dontcallmedom** [comment]
see also https://github.com/w3c/i-slide/issues/35 where @tidoust and I discussed making the i-slide library itself provide a way to use it as an upgrade to existing standard markup

(to use it here, I would prefer using simple `<a href>` links rather than `<iframe>` which would provide a pretty performance-costly fallback)

**Comment by dontcallmedom** [comment]
@bert-github I believe this is now ready for inclusion in the main version of scribe.perl - the script now generates `<a class=islide href=''>` markup that gets automatically replaced by `i-slide` web components upon the invocation of the i-slide module.

When this gets deployed, we should look at documenting it and announcing it to chairs.

**Comment by dontcallmedom** [comment]
(the CI fails for unrelated reasons)

**Pull request merged**